locator/tablets.cc

```diff
@@ -267,7 +314,22 @@ class tablet_effective_replication_map : public effective_replication_map {
     virtual inet_address_vector_replica_set get_natural_endpoints(const token& search_token) const override {
         auto&& tablets = get_tablet_map();
         auto tablet = tablets.get_tablet_id(search_token);
-        auto&& replicas = tablets.get_tablet_info(tablet).replicas;
+        auto* info = tablets.get_tablet_transition_info(tablet);
+        auto&& replicas = std::invoke([&] () -> const tablet_replica_set& {
+            if (!info) {
```

Comment:
In which case there will be no `info`?

Reply:
info is `tablet_transition_info`, which is only present during tablet migration. tablet_info is always present.